scripts/build_kb.py

```diff
@@ -0,0 +1,136 @@
+                for line in root_file.readlines():
+                    line = line.replace('\n', '')
+                    # note: with current implementation, line is considered a comment if it's the first character in the line
+                    if line in paths or line.startswith('#') or re.match(r"^\s*$", line):
+                       continue
+                    else:
+                        absolute_path = abspath(join(root_path, line))
+                        paths.add(absolute_path)
+                        search_knowledge_bases(
+                        absolute_path, output_path, filename)
+        except FileNotFoundError:
+            #this branch will be used when a subsequent KB doesn't have any dependencies
+            pass
+    else:
+        print("Folder", root_path, "is not found.")
+        exit(1)
+
+
+def copy_kb(output_path: str):
+    prepared_kb = join(output_path, "prepared_kb")
+    try:
+        common_prefix = commonprefix(list(paths))
+        for path in paths:
+            # removes the common part from paths and copies what's left to prepared_kb
+            # TODO: ignore .git folder inside kb folders 
+            shutil.copytree(path, join(prepared_kb, relpath(
+                path, common_prefix)), dirs_exist_ok=True)
+    except Exception as e:
+        print(e)
+
+
+def parse_config(path: str) -> dict:
+    config_dict = {'repo_folder': '', 'output_path': '', 'errors_file_path': ''}
+    config = configparser.ConfigParser()
+    config.read(path)
+    config_dict.update({'output_path': abspath(join(dirname(path), config['Repo']['Path']))})
+    config_dict.update({'errors_file_path': abspath(join(dirname(path), config['Repo']['Logfile'], "prepare.log"))})
+
+    return config_dict
+
+
+def prepare_kb(kb_to_prepare: str, logfile: str):
+    for script in prepare_scripts:
+        os.system('python3 ' + script + ' ' + kb_to_prepare + ' ' + logfile)
+    if exists(logfile):
+        print("Error while executing prepare_kb scripts. Check " + logfile + " for logs")
+        exit(1)
+
+
+def build_kb(bin_folder: str, kb_to_build: str):
+    bin_folder = join(bin_folder, "kb.bin")
+    os.makedirs(bin_folder, exist_ok=True)
+    # call sc-builder with required parameters
+    os.system(" ".join([join(ostis_path, "bin/sc-builder"), "-f", "-c", "-i", kb_to_build, "-o", bin_folder, "-e", join(ostis_path, "bin/extensions")]))
+
+
+
+def main(args: dict):
+    conf = parse_config(args["config_file_path"])
+
+    #absolutize paths passed as flags
+
+    #rewrite options which were given by flags
+     
+    for key in args.keys():
+        if args[key] is not None:
+            if key in ["repo_folder", "output_path", "errors_file_path"]: 
+                flag = abspath(join(os.getcwd(), args[key]))
+            else: flag = args[key] 
+            conf[key] = flag
+    # output the final configuration for the script, just to be sure about what's going on.
+    print("args:", conf)
```

Comment:
I think better to use logging lib with different log levels instead of print. But maybe it's needless for script file. What you think about it?